### PR TITLE
Append a short description to branch names

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Update the dev docs to make `MM-$NUMBER_$DES` the new naming convention.

#### Ticket Link
Follow up on https://community-release.mattermost.com/core/pl/6rcj74f9zi8w7ksjgusjehhy9o

### CODE REVIEW

File: site/content/contribute/getting-started/contribution-checklist.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Follow this checklist for submitting a pull request (PR):
 
     5.1 In the webapp repository run `make i18n-extract` to generate the new/updated strings.
 6. The PR is submitted against the Mattermost `master` branch from your fork.
-7. The PR title begins with the Jira or GitHub Ticket ID (e.g. `MM-394` or `GH-394`) and summary template is filled out.
+7. The PR title begins with the Jira or GitHub Ticket ID (e.g. `[MM-394]` or `[GH-394]`) and summary template is filled out.
```
Comment (hanzei) [inline site/content/contribute/getting-started/contribution-checklist.md]: Using braces here is the most common pattern and I updated the docs to reflect that.

File: site/content/contribute/server/developer-workflow.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ Here's a general workflow for a Mattermost developer working on the [mattermost-
     * [./store/](https://github.com/mattermost/mattermost-server/tree/master/store) holds all database querying code.
     * [./utils/](https://github.com/mattermost/mattermost-server/tree/master/utils) holds all utilities, such as the mail utility.
     * [./i18n/](https://github.com/mattermost/mattermost-server/tree/master/i18n) holds all localization files for the server.
-2. On your fork, create a branch `MM-####` where #### is the ticket number if it's a [Jira](https://mattermost.atlassian.net) ticket, or `GH-####` if it's a GitHub Issue without a Jira ticket.
+2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DES` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on. `$DES` is be a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```
Comment (jasonblais) [inline site/content/contribute/server/developer-workflow.md]:
```suggestion
2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DESCRIPTION` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on. `$DESCRIPTION` is be a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```

File: site/content/contribute/webapp/developer-workflow.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ Here's a general workflow a Mattermost developer working on the [mattermost-weba
     * [./i18n/](https://github.com/mattermost/mattermost-webapp/tree/master/i18n) holds the localization files for the client
     * [./utils/](https://github.com/mattermost/mattermost-webapp/tree/master/utils) holds all widely-used utilities
     * [./tests/](https://github.com/mattermost/mattermost-webapp/tree/master/tests) holds all the client unit tests
-2. On your fork, create a branch `MM-####` where #### is the ticket number if it is a [Jira](https://mattermost.atlassian.net) ticket, or `GH-####` if it is a GitHub Issue without a Jira ticket.
+2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DES` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on. `$DES` is be a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```
Comment (jasonblais) [inline site/content/contribute/webapp/developer-workflow.md]:
```suggestion
2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DESCRIPTION` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on. `$DESCRIPTION` is be a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```

File: site/content/contribute/server/developer-workflow.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ Here's a general workflow for a Mattermost developer working on the [mattermost-
     * [./store/](https://github.com/mattermost/mattermost-server/tree/master/store) holds all database querying code.
     * [./utils/](https://github.com/mattermost/mattermost-server/tree/master/utils) holds all utilities, such as the mail utility.
     * [./i18n/](https://github.com/mattermost/mattermost-server/tree/master/i18n) holds all localization files for the server.
-2. On your fork, create a branch `MM-####` where #### is the ticket number if it's a [Jira](https://mattermost.atlassian.net) ticket, or `GH-####` if it's a GitHub Issue without a Jira ticket.
+2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DESCRIPTION` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on. `$DESCRIPTION` is be a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```
Comment (justinegeffen) [inline site/content/contribute/server/developer-workflow.md]:
```suggestion
2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DESCRIPTION` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on and `$DESCRIPTION` is a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```

File: site/content/contribute/webapp/developer-workflow.md [hunk 5]
```diff
@@ -18,7 +18,7 @@ Here's a general workflow a Mattermost developer working on the [mattermost-weba
     * [./i18n/](https://github.com/mattermost/mattermost-webapp/tree/master/i18n) holds the localization files for the client
     * [./utils/](https://github.com/mattermost/mattermost-webapp/tree/master/utils) holds all widely-used utilities
     * [./tests/](https://github.com/mattermost/mattermost-webapp/tree/master/tests) holds all the client unit tests
-2. On your fork, create a branch `MM-####` where #### is the ticket number if it is a [Jira](https://mattermost.atlassian.net) ticket, or `GH-####` if it is a GitHub Issue without a Jira ticket.
+2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DESCRIPTION` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on. `$DESCRIPTION` is be a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```
Comment (justinegeffen) [inline site/content/contribute/webapp/developer-workflow.md]:
```suggestion
2. On your fork, create a feature branch for your changes. Name it `MM-$NUMBER_$DESCRIPTION` where `$NUMBER` is the [Jira](https://mattermost.atlassian.net) ticket number you are working on and `$DESCRIPTION` is a short description of your changes. Example branch names are `MM-18150_plugin-panic-log` and `MM-22037_uppercase-email`.
```

**Pull request merged**